pagopa/io-app
# fix: [IOPID-2490,IOPID-2506] Add logic to validate URLs in Webview and add header to go back on CieID webview screen

## Short description
In order to mitigate security problems, logic was added to check that URLs that are injected into the Webview are in the IPZS domain. 

## List of changes proposed in this pull request
- Add logic to check if URLs are in the IPZS domain
- Add a new screen to redirect a user who obtains a non-compliant URL
- Add mixpanel event in new screen 
- Add tests
- Add logic to show go back button on header

## Demo

### Malicious interaction (unhappy path)
| Android | iOS | 
| - | - | 
| <video src="https://github.com/user-attachments/assets/6610c1b9-c80b-42e9-8925-20a690696338"/> | <video src="https://github.com/user-attachments/assets/968d49a6-d594-4365-b3d3-ef14470403ff"/> | 

### Not Malicious interaction (happy path) + add header to go back during login
| Android (CieID app not configured) | Android (CieID app configured) | iOS (CieID app not configured) |  iOS (CieID app configured) | 
| - | - | - | - |
| <video src="https://github.com/user-attachments/assets/02eae83e-8eb6-4ceb-a117-338573a832b1"/> | <video src="https://github.com/user-attachments/assets/93aeb3da-4191-4e5d-953a-5406f464d5f7"/> | <video src="https://github.com/user-attachments/assets/a748bcd1-1917-4701-a886-3e7ce6c8851a"/> | <video src="https://github.com/user-attachments/assets/91f1b63c-490a-40d1-9c0d-b026c5aa1a81"/> | 

## How to test unhappy path
- run the application using the production environment 
-  if you're using an iOS follow the video and attach in safari this url: `iologincie:https://it.wikipedia.org/wiki/File:Goldfish3.jpg`
- if you're using an Android device you need to install on device a fake cieid app (you can send me a message and i can help you to create and install this application) 

## How to test happy path
- install CieID app
- run IO application using the production environment 
- test as you can see in Demo section
